Doc/c-api/long.rst

```diff
@@ -597,10 +597,176 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
    Exactly what values are considered compact is an implementation detail
    and is subject to change.
 
+   .. versionadded:: 3.12
+
+
 .. c:function:: Py_ssize_t PyUnstable_Long_CompactValue(const PyLongObject* op)
 
    If *op* is compact, as determined by :c:func:`PyUnstable_Long_IsCompact`,
    return its value.
 
    Otherwise, the return value is undefined.
 
+   .. versionadded:: 3.12
+
+
+Export API
+^^^^^^^^^^
+
+.. versionadded:: 3.14
+
+.. c:type:: Py_digit
+
+   A single unsigned digit in the range [``0``; ``PyLong_BASE - 1``].
+
+   It is usually used in an *array of digits*, such as the
+   :c:member:`PyLong_DigitArray.digits` array.
+
+   Its size depend on the :c:macro:`!PYLONG_BITS_IN_DIGIT` macro:
+   see the ``configure`` :option:`--enable-big-digits` option.
+
+   See :c:member:`PyLongLayout.bits_per_digit` for the number of bits per
+   digit and :c:member:`PyLongLayout.digit_size` for the size of a digit (in
+   bytes).
+
+
+.. c:struct:: PyLongLayout
+
+   Layout of an array of digits, used by Python :class:`int` object.
+
+   Use :c:func:`PyLong_GetNativeLayout` to get the native layout of Python
+   :class:`int` objects.
+
+   See also :attr:`sys.int_info` which exposes similar information to Python.
+
+   .. c:member:: uint8_t bits_per_digit
+
+      Bits per digit.
+
+   .. c:member:: uint8_t digit_size
+
+      Digit size in bytes.
+
+   .. c:member:: int8_t digits_order
+
+      Word endian:
+
+      - ``1`` for most significant word first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+   .. c:member:: int8_t endian
+
+      Array endian:
+
+      - ``1`` for most significant byte first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+
+.. c:function:: const PyLongLayout* PyLong_GetNativeLayout(void)
+
+   Get the native layout of Python :class:`int` objects.
+
+   See the :c:struct:`PyLongLayout` structure.
+
+
+.. c:struct:: PyLong_DigitArray
+
+   A Python :class:`int` object exported as an array of digits.
+
+   .. c:member:: int negative
+
+      1 if the number is negative, 0 otherwise.
+
+   .. c:member:: Py_ssize_t ndigits
+
+      Number of digits in :c:member:`digits` array.
+
+   .. c:member:: const void *digits
+
+      Read-only array of unsigned digits.
+
+   .. c:member:: Py_uintptr_t reserved
```

Comment:
Should we document internal fields at all?

Reply:
I removed the documentation of this member.